src/js/util/Topology.js

```diff
@@ -262,13 +263,32 @@ var Topology = function(params, offsets){
         }
 	
         data.links = links;
+        //Fix for the endpoints bug. 
+        //make endpoints appear on all link endings. 
+
+        if(!data.endpoints) data.endpoints = [];  
+        var tempEndPointsArray = data.endpoints.slice();   
+        
+        data.endpoints = []; 
+        _.forEach(data.links, function(link){
+            _.forEach(link.path, function(waypoint){
+                if(waypoint.endpoint == true){
+                    var endpointObj = {}; 
+                    endpointObj.lon = waypoint.lon; 
+                    endpointObj.lat = waypoint.lat; 
+                    endpointObj.name = tempEndPointsArray.find(obj => {
+                        return obj.lat === waypoint.lat; 
+                      }).name;
```

Comment:
This fails for the case when tempEndPointsArray is empty since we are trying to chain .name() on .find() 
.find() will never execute when the array is empty and therefore calling .name() would throw an error.

Reply:
I will do it as soon as possible. Thanks